dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java

```diff
@@ -198,6 +198,34 @@ public void testWorkflowLaunchOrNotLaunchBasedOnCredentials() throws IOException
         Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file.txt"), "workflow", "launch", "--entry", toolpath, "--json" , ResourceHelpers.resourceFilePath("md5sum-wrapper-tool.json") ,  "--script" });
     }
 
+    /**
+     * This tests workflow convert entry2json when the main descriptor is nested far within the GitHub repo with secondary descriptors too
+     * @throws IOException
+     */
+    @Test
+    public void testEntryConvertWDLWithSecondaryDescriptors() throws IOException {
+        String toolpath = SourceControl.GITHUB.toString() + "/garyluu/skylab";
+        final ApiClient webClient = getWebClient();
+        WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+        Workflow workflow = workflowApi
+                .manualRegister(SourceControl.GITHUB.getFriendlyName(), "garyluu/skylab", "/pipelines/smartseq2_single_sample/SmartSeq2SingleSample.wdl",
+                        "", "wdl", null);
+        Workflow refresh = workflowApi.refresh(workflow.getId());
+        final PublishRequest publishRequest = SwaggerUtility.createPublishRequest(true);
+        workflowApi.publish(refresh.getId(), publishRequest);
+
+        Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "workflow", "convert", "entry2json", "--entry", toolpath + ":Dockstore_Testing", "--script" });
+        Assert.assertTrue(systemOutRule.getLog().contains("{\n" + "  \"SmartSeq2SingleCell.gtf_file\": \"File\",\n"
```

Comment:
This is getting a bit out of hand and will break if the file output format changes with different newlines or spacing.